staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -176,13 +176,94 @@ type Histogram struct {
 	*dto.Histogram
 }
 
-// GetHistogramFromGatherer collects a metric from a gatherer implementing k8s.io/component-base/metrics.Gatherer interface.
+// HistogramVec wraps a slice of Histogram.
+// Note that each Histogram must have the same number of buckets.
+type HistogramVec []*Histogram
+
+// GetSampleCount aggregates the sample count of each inner Historgram.
+func (vec HistogramVec) GetSampleCount() uint64 {
+	var count uint64
+	for _, hist := range vec {
+		count += hist.GetSampleCount()
+	}
+	return count
+}
+
+// GetSampleSum aggregates the sample sum of each inner Historgram.
+func (vec HistogramVec) GetSampleSum() float64 {
+	var sum float64
+	for _, hist := range vec {
+		sum += hist.GetSampleSum()
+	}
+	return sum
+}
+
+// Quantile first aggregates inner buckets of each Histogram, and then
+// computes q-th quantile of a cumulative histogram.
+func (vec HistogramVec) Quantile(q float64) float64 {
+	var buckets []bucket
+
+	for _, hist := range vec {
+		for i, bckt := range hist.Bucket {
+			if len(buckets) <= i {
+				buckets = append(buckets, bucket{
+					count:      float64(bckt.GetCumulativeCount()),
+					upperBound: bckt.GetUpperBound(),
+				})
+			} else {
+				buckets[i].count += float64(bckt.GetCumulativeCount())
+			}
+		}
+	}
+
+	if len(buckets) == 0 || buckets[len(buckets)-1].upperBound != math.Inf(+1) {
+		// The list of buckets in dto.Histogram doesn't include the final +Inf bucket, so we
+		// add it here for the reset of the samples.
```

Comment:
I assume you meant "rest of the samples". If so, wouldn't `count: float64(vec.GetSampleCount())` include ALL of the samples?

Reply:
Yes, it's intentional to include all samples. Its previous buckets aggregate bucket from each histogram, and hence the same to this last bucket.